### PR TITLE
Feature/pdct 1396 flatten out template json

### PR DESCRIPTION
**Pull request by annaCPR**
# Description
- updates the json template for ingest to have a new flat structure with collections, families, documents and events all as top level members
- updates relevant tests to reflect this change

Please include:

- a summary of the changes
- links to any related issue/ticket
- any additional relevant motivation and context
- details of any dependency updates that are required for this change

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [x] The PR represents a single feature (small drive-by fixes are also ok)
- [x] The PR includes tests that are sufficient for the level of risk
- [x] The code is sufficiently commented, particularly in hard-to-understand areas
- [x] Any required documentation updates have been made
- [x] Any TODOs added are captured in future tickets
- [x] No FIXMEs remain

### CODE REVIEW

File: app/api/api_v1/routers/ingest.py [hunk 1]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
 
 
-def get_metadata_template(corpus_type: str):
-    return taxonomy.get(corpus_type)
+def get_metadata_template(corpus_type: str, metadata_type: str):
+    metadata = taxonomy.get(corpus_type)
+    if not metadata:
+        return {}
+    if metadata_type == "document":
+        return metadata.pop("_document")
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:50]: Could we use an Enum here? (see db-client [EntitySpecificTaxonomyKeys](https://github.com/climatepolicyradar/navigator-db-client/blob/c1d6e7f8fead91df449648307550b9a4ad63a45b/db_client/models/dfce/taxonomy_entry.py#L29))

File: app/api/api_v1/routers/ingest.py [hunk 2]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
+    elif metadata_type == "family":
+        metadata.pop("_document")
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:52]: Could we use an Enum here? (see db-client [EntitySpecificTaxonomyKeys](https://github.com/climatepolicyradar/navigator-db-client/blob/c1d6e7f8fead91df449648307550b9a4ad63a45b/db_client/models/dfce/taxonomy_entry.py#L29))

File: app/api/api_v1/routers/ingest.py [hunk 3]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
 
-def get_metadata_template(corpus_type: str):
-    return taxonomy.get(corpus_type)
+def get_metadata_template(corpus_type: str, metadata_type: str):
+    metadata = taxonomy.get(corpus_type)
+    if not metadata:
+        return {}
+    if metadata_type == "document":
+        return metadata.pop("_document")
+    elif metadata_type == "family":
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:51]: Could we use an Enum here for `family`? (see db-client [CountedEntity](https://github.com/climatepolicyradar/navigator-db-client/blob/c1d6e7f8fead91df449648307550b9a4ad63a45b/db_client/models/organisation/counters.py#L31))

File: app/api/api_v1/routers/ingest.py [hunk 4]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
     return document_template
 
 
-def get_metadata_template(corpus_type: str):
-    return taxonomy.get(corpus_type)
+def get_metadata_template(corpus_type: str, metadata_type: str):
+    metadata = taxonomy.get(corpus_type)
+    if not metadata:
+        return {}
+    if metadata_type == "document":
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:49]: Could we use an Enum here for `document`, e.g., by the value of metadata_type is either the str value corresponding to the Enum or the enum reference itself? (see db-client [CountedEntity](https://github.com/climatepolicyradar/navigator-db-client/blob/c1d6e7f8fead91df449648307550b9a4ad63a45b/db_client/models/organisation/counters.py#L31))

File: app/api/api_v1/routers/ingest.py [hunk 5]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
-    del document_template["family_import_id"]
-    document_template["events"] = [get_event_template()]
+    document_template["metadata"] = get_metadata_template(corpus_type, "document")
 
     return document_template
 
 
-def get_metadata_template(corpus_type: str):
-    return taxonomy.get(corpus_type)
+def get_metadata_template(corpus_type: str, metadata_type: str):
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:45]: What's the return type for this function?

File: app/api/api_v1/routers/ingest.py [hunk 6]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
     event_template = event_schema["properties"]
 
-    del event_template["family_document_import_id"]
-    del event_template["family_import_id"]
-
     return event_template
 
 
-def get_document_template():
+def get_document_template(corpus_type: str):
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:37]: What's the return type for this function?

File: app/api/api_v1/routers/ingest.py [hunk 7]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
+    if not metadata:
+        return {}
+    if metadata_type == "document":
+        return metadata.pop("_document")
+    elif metadata_type == "family":
+        metadata.pop("_document")
+    return metadata
 
 
 def get_family_template(corpus_type: str):
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:56]: What's the return type for this function?

Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:56]: What's the return type for this function?

File: app/api/api_v1/routers/ingest.py [hunk 8]
```diff
@@ -57,23 +59,9 @@ def get_family_template(corpus_type: str):
 
     del family_template["corpus_import_id"]
 
-    # look up taxonomy by corpus type
-    family_metadata = get_metadata_template(corpus_type)
-    # pull out document taxonomy
-    document_metadata = family_metadata.pop("_document") if family_metadata else {}
-
-    # add family metadata and event templates to the family template
+    family_metadata = get_metadata_template(corpus_type, "family")
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:62]: Potential enum use

File: app/api/api_v1/routers/ingest.py [hunk 9]
```diff
@@ -31,24 +31,26 @@ def get_event_template():
...
 
 
-def get_document_template():
+def get_document_template(corpus_type: str):
     document_schema = IngestDocumentDTO.model_json_schema(mode="serialization")
     document_template = document_schema["properties"]
-
-    del document_template["family_import_id"]
-    document_template["events"] = [get_event_template()]
+    document_template["metadata"] = get_metadata_template(corpus_type, "document")
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:40]: Potential enum use

File: app/api/api_v1/routers/ingest.py (hunk 6 above)
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:37]: Could we add a docstring please? 😄

File: app/api/api_v1/routers/ingest.py (hunk 5 above)
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:45]: Could we add a docstring please? 😄

File: app/api/api_v1/routers/ingest.py (hunk 7 above)
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:56]: Could we add a docstring please? 😄

**Review by katybaulch — CHANGES_REQUESTED** [review]
logically all good - just a few code quality improvements that can be made :)

File: app/api/api_v1/routers/ingest.py [hunk 10]
```diff
@@ -20,60 +22,75 @@
...
 
     return document_template
 
 
-def get_metadata_template(corpus_type: str):
-    return taxonomy.get(corpus_type)
+def get_metadata_template(corpus_type: str, metadata_type: CountedEntity) -> dict:
+    """
+    Gets a metadata template for a given corpus type and entity.
+
```
Comment (katybaulch) [inline app/api/api_v1/routers/ingest.py:67]: Missing :param lines for corpus_type and metadata_type - please see [here](https://github.com/climatepolicyradar/navigator-admin-backend/blob/2674536196424175743efe20d72af0cc8cd2942f/app/repository/document.py#L201) for example

**Review by katybaulch — CHANGES_REQUESTED** [review]
Still missing docstring parameter information, nearly there though - well done! ✨

File: app/api/api_v1/routers/ingest.py [hunk 11]
```diff
@@ -20,60 +22,75 @@
...
+    :return dict: The metadata template.
+    """
+    metadata = taxonomy.get(corpus_type)
+    if not metadata:
+        return {}
+    if metadata_type == CountedEntity.Document:
+        return metadata.pop(EntitySpecificTaxonomyKeys.DOCUMENT.value)
+    elif metadata_type == CountedEntity.Family:
+        metadata.pop(EntitySpecificTaxonomyKeys.DOCUMENT.value)
```
[the remark this answers is not in the document]
Comment (annaCPR) [reply app/api/api_v1/routers/ingest.py:79]: Double checked but unfortunately it's correct. The _document part is embedded within the larger family taxonomy and there is no "family" key as such. So a family taxonomy is essentially taxonomy - _document. But if you can think of a more elegant way to refactor this let me know cos I do think it's a bit clunky.

[the remark this answers is not in the document]
Comment (katybaulch) [reply app/api/api_v1/routers/ingest.py:79]: Is the family taxonomy is not then `taxonomy - _document - event_type` (sorry, out of the loop here!)?

**Pull request merged**